File: scala-example/spring-boot-example/src/main/scala/com/github/poad/scala/springboot/example/entity/MessageEntity.scala
package com.github.poad.scala.springboot.example.entity

import javax.persistence.{Column, Entity, Id, Table}

import scala.beans.BeanProperty

@Entity
@Table(name="message")
case class MessageEntity(@BeanProperty @Column(name="message") var message: String) extends Serializable {
  // default constructor for JPA
  def this() {
    this(null)
  }

  @Id
  @Column(name="id")
  @BeanProperty
  var id: String = _
}

File: java-example/spring-boot/spring-boot-jpa-example/src/main/java/com/github/poad/example/springboot/jpa/entity/MessageEntity.java
package com.github.poad.example.springboot.jpa.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.PrimaryKeyJoinColumn;
import jakarta.persistence.Table;

/**
 * Created by ken-yo on 2016/08/06.
 */
@Entity
@Table(name = "message")
@PrimaryKeyJoinColumn(name = "id")
public class MessageEntity extends Identifier {
    @Column(name = "name", nullable = true)
    private String name;

    @Column(name = "message")
    private String message;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
